test/sam.c

```diff
@@ -1043,6 +1043,41 @@ static void faidx1(const char *filename)
     fai_destroy(fai);
 }
 
+static void test_empty_sam_file(const char *filename)
+{
+    samFile *in = sam_open(filename, "r");
+    if (in) {
+        bam1_t *aln = bam_init1();
```

Comment:
A check for the _empty_format_ would be a nice addition.
Something like:
```c
if (in->format.format != empty_format)
    fail("file %s is not marked as empty\n", filename);
```

Reply:
Good idea, thanks.